common/middleware/ensure-authenticated.js

```diff
@@ -9,13 +9,33 @@ function _isExpired(authExpiry) {
 module.exports = function ensureAuthenticated({
   provider,
   whitelist = [],
+  expiryMargin = 5 * 60,
 } = {}) {
   return (req, res, next) => {
-    if (whitelist.includes(req.url) || !_isExpired(req.session.authExpiry)) {
+    let authExpiry = req.session.authExpiry
+    if (req.method === 'GET') {
+      authExpiry -= expiryMargin
+    }
+    if (whitelist.includes(req.url) || !_isExpired(authExpiry)) {
       return next()
     }
 
     req.session.originalRequestUrl = req.originalUrl
+    if (req.method === 'POST') {
+      let errorKey
+      const contentType = req.header('content-type')
+      if (contentType.startsWith('multipart/form-data')) {
+        errorKey = 'MULTIPART_FAILED_AUTH'
+      } else if (req.xhr) {
+        errorKey = 'DELETE_FAILED_AUTH'
+      }
+      if (errorKey) {
+        res.status(401)
+        const errorString = req.t(`validation::${errorKey}`)
+        return res.send(errorString)
+      }
+      req.session.originalRequestBody = req.body
```

Comment:
Could this cause us any issues if it becomes quite large?

Reply:
We're never storing the file uploads, so how large are you expecting a standard POST to grow to? Surely if it managed to be post as a regular www/url-encoded form, we won't have issues stashing it in the redis db?